src/Scripts/FSScriptLoader.spec.ts

```diff
@@ -0,0 +1,80 @@
+import 'reflect-metadata';
+import { DefaultVirtualScripts } from './DefaultVirtualScripts';
+import { FSScriptLoader } from './FSScriptLoader';
+import { VirtualScriptType } from './VirtualScript';
+import { logger } from '../Utils';
+import {
+  anyFunction,
+  anyString,
+  anything,
+  instance,
+  mock,
+  spy,
+  verify,
+  when
+} from 'ts-mockito';
+import { expect, use } from 'chai';
+import promisified from 'chai-as-promised';
+import fs from 'fs';
+
+use(promisified);
+
+describe('FSScriptLoader', () => {
+  const createScriptLoader = () => {
+    const mockedVirtualScripts = mock(DefaultVirtualScripts);
+    const virtualScripts = instance(mockedVirtualScripts);
+    const scriptLoader = new FSScriptLoader(virtualScripts);
+
+    return { mockedVirtualScripts, scriptLoader };
+  };
+
+  const spiedFs = spy(fs);
```

Comment:
* Use `beforeEach` and `afterEach` to instantiate a SUT and reset mocks. 
* The `createScriptLoader` function is redundant.
* Create mocks for interfaces, not implementations
 
```suggestion
  const mockedVirtualScripts = mock<VirtualScripts>();
  const spiedFs = spy(fs);
  
  let scriptLoader!: FSScriptLoader;
  
  beforeEach(() => {
    scriptLoader = new FSScriptLoader(instance(mockedVirtualScripts));
  });
  
  afterEach(() => reset<unknown>(mockedVirtualScripts, spiedFs));
```